src/partition.c

```diff
@@ -45,12 +45,22 @@
 #ifdef WITH_MPI
 #include <mpi.h>
 /* METIS/ParMETIS headers only used when MPI is also available. */
-#ifdef HAVE_PARMETIS
+#if defined(HAVE_PARMETIS) && !defined(HAVE_SCOTCH)
 #include <parmetis.h>
 #endif
-#ifdef HAVE_METIS
+#if defined(HAVE_METIS) && !defined(HAVE_SCOTCH)
 #include <metis.h>
 #endif
+/* SCOTCH headers only used when MPI is also available. */
+#ifdef HAVE_SCOTCH
+#include <scotch.h>
+typedef int64_t idx_t;
+#define IDX_T MPI_INT
```

Comment:
These two lines are wrong or not used (not used I suspect). The sizes of these should match 32 or 64 bits
as used by the library. Remove them.

Reply:
Agreed. I have removed lines 57 and 58 and no errors came up. This # ifdef was outside an # ifdef HAVE_MPI before, so the CPPFLAGS in the Makefile was missing SCOTCH_INCS and I had errors / breaks in the build. After moving it inside HAVE_MPI fence, it was OK, but they turn out to be redundant. Lines 57 and 58 will not be present in the prod MR.